[java] ModifierOrder: wrong enum values documented, indirectly causing xml parse errors

**Affects PMD Version:** 7.17.0 

**Description:**

In `ModifierOrderRule` for the `typeAnnotations` property the `TypeAnnotationPosition` enum is used. The [ModifierOrder docs](https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#modifierorder) show as possible values: `on type`, `on decl` and `anywhere` (default). Note the space is the first two values.

When using these as values in a custom override, there is an xml exception during parsing:

```
[ERROR] Error at /.../pmd-main.xml:196:40
 194|     <priority>3</priority>
 195|     <properties>
 196|       <property name="typeAnnotations" value="on decl" />
                                             ^^^^^ Value should be one of ''ontype'', ''anywhere'', ''ondecl''
```

In the enum there is a `label()` to translate to `ontype` and `ondecl` (without space).

**Exception Stacktrace:**

```
java.lang.IllegalStateException: net.sourceforge.pmd.lang.rule.RuleSetLoadException: Cannot load ruleset /.../pmd-main.xml: 2 XML validation errors occurred
	at org.sonar.plugins.pmd.AbstractPmdExecutor.createRuleSet(AbstractPmdExecutor.java:202)
	at org.sonar.plugins.pmd.AbstractPmdExecutor.executeRules(AbstractPmdExecutor.java:176)
	at org.sonar.plugins.pmd.PmdJavaExecutor.executePmd(PmdJavaExecutor.java:68)
	at org.sonar.plugins.pmd.AbstractPmdExecutor.execute(AbstractPmdExecutor.java:90)
	at org.sonar.plugins.pmd.PmdSensor.execute(PmdSensor.java:97)
	at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:69)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:88)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.lambda$execute$1(ModuleSensorsExecutor.java:61)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.withModuleStrategy(ModuleSensorsExecutor.java:79)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:61)
	at org.sonar.scanner.scan.SpringModuleScanContainer.doAfterStart(SpringModuleScanContainer.java:80)
	at org.sonar.core.platform.SpringComponentContainer.startComponents(SpringComponentContainer.java:227)
	at org.sonar.core.platform.SpringComponentContainer.execute(SpringComponentContainer.java:206)
	at org.sonar.scanner.scan.SpringProjectScanContainer.scan(SpringProjectScanContainer.java:212)
	at org.sonar.scanner.scan.SpringProjectScanContainer.scanRecursively(SpringProjectScanContainer.java:208)
	at org.sonar.scanner.scan.SpringProjectScanContainer.doAfterStart(SpringProjectScanContainer.java:178)
	at org.sonar.core.platform.SpringComponentContainer.startComponents(SpringComponentContainer.java:227)
	at org.sonar.core.platform.SpringComponentContainer.execute(SpringComponentContainer.java:206)
	at org.sonar.scanner.bootstrap.SpringScannerContainer.doAfterStart(SpringScannerContainer.java:339)
	at org.sonar.core.platform.SpringComponentContainer.startComponents(SpringComponentContainer.java:227)
	at org.sonar.core.platform.SpringComponentContainer.execute(SpringComponentContainer.java:206)
	at org.sonar.scanner.bootstrap.SpringGlobalContainer.doAfterStart(SpringGlobalContainer.java:142)
	at org.sonar.core.platform.SpringComponentContainer.startComponents(SpringComponentContainer.java:227)
	at org.sonar.core.platform.SpringComponentContainer.execute(SpringComponentContainer.java:206)
	at org.sonar.scanner.bootstrap.ScannerMain.runScannerEngine(ScannerMain.java:157)
	at org.sonar.scanner.bootstrap.ScannerMain.run(ScannerMain.java:72)
	at org.sonar.scanner.bootstrap.ScannerMain.main(ScannerMain.java:56)
Caused by: net.sourceforge.pmd.lang.rule.RuleSetLoadException: Cannot load ruleset /.../pmd-main.xml: 2 XML validation errors occurred
	at net.sourceforge.pmd.lang.rule.RuleSetFactory.readDocument(RuleSetFactory.java:196)
	at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:119)
	at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:114)
	at net.sourceforge.pmd.lang.rule.RuleSetLoader.loadFromResource(RuleSetLoader.java:248)
	at net.sourceforge.pmd.lang.rule.RuleSetLoader.loadFromResource(RuleSetLoader.java:139)
	at org.sonar.plugins.pmd.AbstractPmdExecutor.parseRuleSetWithReporter(AbstractPmdExecutor.java:216)
	at org.sonar.plugins.pmd.AbstractPmdExecutor.createRuleSet(AbstractPmdExecutor.java:200)
	... 26 common frames omitted
```

**Code Sample demonstrating the issue:**

```xml
  <rule ref="category/java/codestyle.xml/ModifierOrder">
    <priority>3</priority>
    <properties>
      <property name="typeAnnotations" value="on decl" />
    </properties>
  </rule>
```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1.  `pmd-bin-7.17.0/bin/pmd check -v -R /.../pmd-main.xml -d .`

**Running PMD through:** *CLI | Maven*

**Note:** 

Enums for properties seem not consistent in the display/label approach: other enums have `toString()` and `getDisplayName()` and also a comment that `RuleDocGenerator` uses the `toString()`. E.g. `net.sourceforge.pmd.lang.java.rule.bestpractices.AvoidReassigningLoopVariablesRule.ForeachReassignOption`. This might be the way for all enums?

Enum `net.sourceforge.pmd.lang.java.rule.bestpractices.AvoidUsingHardCodedIPRule.AddressKinds` also has a `label`, but implemented a bit different.

--- a/docs/pages/release_notes.md
+++ b/docs/pages/release_notes.md
@@ -92,6 +92,7 @@ checkstyle version during the build.
   * [#6057](https://github.com/pmd/pmd/issues/6057): \[java] ModifierOrder false positive on "abstract sealed class"
   * [#6079](https://github.com/pmd/pmd/issues/6079): \[java] IdenticalCatchBranches: False negative for overriden method calls
   * [#6123](https://github.com/pmd/pmd/issues/6123): \[java] UselessParentheses FP around switch expression
+  * [#6131](https://github.com/pmd/pmd/issues/6131): \[java] ModifierOrder: wrong enum values documented, indirectly causing xml parse errors
 * java-design
   * [#1499](https://github.com/pmd/pmd/issues/1499): \[java] AvoidDeeplyNestedIfStmts violations can be unintentionally undetected
   * [#5569](https://github.com/pmd/pmd/issues/5569): \[java] ExcessivePublicCount should report number of public "things"

--- a/pmd-java/src/main/resources/category/java/codestyle.xml
+++ b/pmd-java/src/main/resources/category/java/codestyle.xml
@@ -1147,8 +1147,8 @@ public class Foo {
             By setting the property `typeAnnotations`, you can also enforce that type
             annotations appear right of the modifier keywords, next to the type they apply to.
             This property can have three values:
-            - `on type`: Type annotations must be placed next to the type they apply to
-            - `on decl`: Type annotations must be placed with other annotations, before modifiers.
+            - `ontype`: Type annotations must be placed next to the type they apply to
+            - `ondecl`: Type annotations must be placed with other annotations, before modifiers.
             This is not enforced if the type annotations syntactically appears within the type, e.g.
             in `public Map.@Nullable Entry&lt;K,V&gt; method()`.
             - `anywhere` (default): Either position fits. They still cannot be interspersed within keyword

